modules/account-lib/test/unit/coin/algo/transactionBuilder/keyRegistrationBuilder.ts

```diff
@@ -0,0 +1,286 @@
+  }
+}
+
+describe('Algo KeyRegistration Builder', () => {
+  let builder: StubTransactionBuilder;
+  
+  const sender = AlgoResources.accounts.account1;
+  const { rawTx } = AlgoResources
+
+
+  beforeEach(() => {
+    const config = coins.get('algo');
+    builder = new StubTransactionBuilder(config)
+  });
+
+  describe('setter validation', () => {
+    it('should validate voteKey, is set and is a valid string', () => {
+      const spy = sinon.spy(builder, 'voteKey');
+      should.doesNotThrow(() => builder.voteKey(sender.voteKey));
+      assert.calledOnce(spy);
+    });
+
+    it('should validate selection key, is set and is a valid string', () => {
+      const spy = sinon.spy(builder, 'selectionKey');
+      should.doesNotThrow(() => builder.selectionKey(sender.selectionKey));
+      assert.calledOnce(spy);
+    });
+
+    it('should validate vote First key is gt than 0', () => {
+      const spy = sinon.spy(builder, 'validateValue');
+      should.throws(
+        () => builder.voteFirst(-1),
+        (e: Error) => e.message === 'Value cannot be less than zero',
+      );
+      should.doesNotThrow(() => builder.voteFirst(15));
+      assert.calledTwice(spy);
+    });
+
+    it('should validate voteFirst set before voteLast', () => {
+      const validateValueSpy = sinon.spy(builder, 'validateValue');
+      const spy = sinon.spy(builder, 'voteLast');
+      should.throws(
+        () => builder.voteLast(10),
+        (e: Error) => e.message === 'VoteKey first round must be set before setting last round',
+      );
+      assert.calledOnce(validateValueSpy)
+      assert.calledOnce(spy);
+    });
+
+    it('should validate vote Last key is gt than 0', () => {
+      const validateValueSpy = sinon.spy(builder, 'validateValue');
+      const spy = sinon.spy(builder, 'voteLast');
+      builder.voteFirst(1);
+      should.throws(
+        () => builder.voteLast(-1),
+        (e: Error) => e.message === 'Value cannot be less than zero',
+      );
+      should.doesNotThrow(() => builder.voteLast(15));
+      assert.calledThrice(validateValueSpy)
+      assert.calledTwice(spy);
+    });
+
+    it('should validate vote Last key is gt than FirstKey', () => {
+      const spy = sinon.spy(builder, 'voteLast');
+      builder.voteFirst(15);
+      should.throws(
+        () => builder.voteLast(12),
+        (e: Error) => e.message === 'VoteKey last round must be greater than first round',
+      );
+      should.doesNotThrow(() => builder.voteLast(18));
+      assert.calledTwice(spy);
+    });
+
+    it('should validate vote Key Dilution', () => {
+      const validateValueSpy = sinon.spy(builder, 'validateValue');
+      const spy = sinon.spy(builder, 'voteKeyDilution');
+      builder
+        .voteFirst(5)
+        .voteLast(18);
+      should.throws(
+        () => builder.voteKeyDilution(25),
+        (e: Error) => e.message === 'Key dilution value must be less than or equal to the square root of the voteKey validity range.',
```

Comment:
Consider using a custom error type here so the test wont be flaky if the message changes